metadata_synthesizer.py

```diff
@@ -64,15 +70,35 @@ def create_mixtures(self):
             nb_rooms_nf = len(rooms_nf)
             
             
-            idx_active = np.array([])
-            for na in range(self._nb_active_classes):
-                idx_active = np.append(idx_active, np.nonzero(self._db_config._samplelist[nfold]['class'] == self._active_classes[na]))
-            idx_active = idx_active.astype('int')
 
-            foldlist_nff['class'] = self._db_config._samplelist[nfold]['class'][idx_active]
-            foldlist_nff['audiofile'] = self._db_config._samplelist[nfold]['audiofile'][idx_active]
-            foldlist_nff['duration'] = self._db_config._samplelist[nfold]['duration'][idx_active]
-            foldlist_nff['onoffset'] = self._db_config._samplelist[nfold]['onoffset'][idx_active]
+            # Load fsd50k object containing dcase to fsd50k paths
+            fsd50k = prepare_fsd50k()
+            foldname = self._foldnames[nfold]
+            files_per_class = fsd50k.get_filenames(foldname)
+            audiofiles = []
+            audioclasses = []
+            durations = []
+            onoffsets = []
+            for k,v in self._classname_dict.items():
+                for kk,vv in files_per_class.items():
+                    if k == kk:
+                        audiofiles.extend(list(vv.values()))
+                        audioclasses.extend([v]*len(vv))
+            audiofiles_ = []
+            for audiof in audiofiles:
+                if type(audiof) == list:

```

Comment:
this is very awkward because we are keeping track of the duration of music files in addition to their path. Let's just keep the file path. For the time bring I think we should instead trim the music files (extra pre-processing step) and simply get their duration using `librosa.get_duration`. Alternatively we can add a duration during loading with `librosa.load`.

Reply:
Thanks @iranroman, after your changes things are much clearer. I added a small change to retrieve the tracks per classes in a single loop.

Reply:
let's use these lines to standardize the durations to a fixed precision
```
for i in range(len(durations)): 
    durations[i] = np.floor(durations[i]*100)/100 
    onoffsets[i][1] = np.floor(onoffsets[i][1]*100)/100 
```